### PR TITLE
Change the toolbar button icon based on theme color

### LINKED ISSUE #719: Test pilot icon appears white
Just installed test pilot add-on on latest nightly, OS X, default theme:

white-button

I already had activity-stream and universal-search installed, nothing peculiar in the browser console.

### PR DESCRIPTION
**Pull request by dannycoates**
this should fix #719 

p.s. I also snuck in changing it to a ToggleButton 🍌

### CODE REVIEW

File: addon/lib/toolbar-button.js [hunk 1]
```diff
@@ -23,11 +26,22 @@ let settings;
 let button;
 let panel;
 let collapsed;
+let prefs;
 
-function setActionButton(dark) {
-  const iconPrefix = dark ? './icon-inverted' : './icon';
+function themeChanged() {
+  setTimeout(setButton);
```
Comment (dannycoates) [inline addon/lib/toolbar-button.js]: weirdly, without the `setTimeout` it sometimes (always?) picks up the style _before_ the change is applied

File: addon/lib/toolbar-button.js [hunk 2]
```diff
@@ -23,11 +26,22 @@ let settings;
...
+
+function isLightTheme() {
+  var chromeWindow = viewFor(windows.activeWindow);
+  return !chromeWindow.document.getElementById('nav-bar').matches('[brighttext]');
+}
 
-  button = ActionButton({ // eslint-disable-line new-cap
+function setButton() {
+  const iconPrefix = isLightTheme() ? './icon' : './icon-inverted';
+  if (button) { button.destroy(); }
```
Comment (dannycoates) [inline addon/lib/toolbar-button.js]: the console was complaining about something already existing when `setButton` ran a second time. This `destroy` made it go away.

**Comment by dannycoates** [comment]
This doesn't work yet when the button is moved into the hamburger menu. I'm looking into it now

**Comment by dannycoates** [comment]
> This doesn't work yet when the button is moved into the hamburger menu

It sort of works now. It ends up with the right shade, but while in customize mode its wrong for dark themes. IDK if there's much else I can do without using an overlay...

**Comment by lmorchard** [comment]
Seems to work for me

**Pull request merged**